test/client_reset_test.dart

```diff
@@ -323,50 +333,58 @@ Future<void> main([List<String>? args]) async {
 
       await triggerClientReset(realm, restartSession: false);
       realm.syncSession.resume();
-      await waitFutureWithTimeout(onAfterCompleter.future, timeoutError: "Neither onAfterDiscard, onAfterDiscard nor onManualResetFallback is reported.");
+      await onAfterCompleter.future.wait(defaultWaitTimeout, "Neither onAfterDiscard, onAfterDiscard nor onManualResetFallback is reported.");
     });
   }
 
-  baasTest('Disabled server recovery - onAfterDiscard callback is invoked for RecoverOrDiscardUnsyncedChangesHandler', (appConfig) async {
-    final app = App(appConfig);
-    final user = await getIntegrationUser(app);
+  {
+    final baasAppName = AppNames.flexible;
+    baasTest('Disabled server recovery - onAfterDiscard callback is invoked for RecoverOrDiscardUnsyncedChangesHandler', appName: baasAppName,
+        (appConfig) async {
+      final app = App(appConfig);
+      final user = await getIntegrationUser(app);
 
-    final onBeforeCompleter = Completer<void>();
-    final onAfterCompleter = Completer<void>();
-    bool recovery = false;
-    bool discard = false;
-
-    final config = Configuration.flexibleSync(user, [Task.schema, Schedule.schema],
-        clientResetHandler: RecoverOrDiscardUnsyncedChangesHandler(
-          onBeforeReset: (beforeResetRealm) => onBeforeCompleter.complete(),
-          onAfterRecovery: (Realm beforeResetRealm, Realm afterResetRealm) {
-            onAfterCompleter.complete();
-            recovery = true;
-          },
-          onAfterDiscard: (Realm beforeResetRealm, Realm afterResetRealm) {
-            onAfterCompleter.complete();
-            discard = true;
-          },
-        ));
+      final onBeforeCompleter = Completer<void>();
+      final onAfterCompleter = Completer<void>();
+      bool recovery = false;
+      bool discard = false;
 
-    final realm = await getRealmAsync(config);
-    await realm.syncSession.waitForUpload();
+      final config = Configuration.flexibleSync(user, [Task.schema, Schedule.schema],
+          clientResetHandler: RecoverOrDiscardUnsyncedChangesHandler(
+            onBeforeReset: (beforeResetRealm) => onBeforeCompleter.complete(),
+            onAfterRecovery: (Realm beforeResetRealm, Realm afterResetRealm) {
+              onAfterCompleter.complete();
+              recovery = true;
+            },
+            onAfterDiscard: (Realm beforeResetRealm, Realm afterResetRealm) {
+              onAfterCompleter.complete();
+              discard = true;
+            },
+          ));
 
-    await disableAutomaticRecovery();
-    await triggerClientReset(realm);
+      final realm = await getRealmAsync(config);
+      await realm.syncSession.waitForUpload();
 
-    await waitFutureWithTimeout(onBeforeCompleter.future, timeoutError: "onBeforeReset is not reported.");
-    await waitFutureWithTimeout(onAfterCompleter.future, timeoutError: "Neither onAfterRecovery nor onAfterDiscard is reported.");
+      await disableAutoRecoveryForApp(baasAppName);
```

Comment:
I would add `disableAutoRecoveryForApp` in the `try` block.

Reply:
If you do that and it fails then the finally block will kick trying to enable auto recovery when in fact the disable of the auto recovery failed.

Reply:
I think that enabling will pass successfully even though the recovery wasn't disabled. According to me it is better to try to enable the recovery in case of some partial disabling has happen, if it is possible.